components/event/org.wso2.carbon.event.admin/src/main/java/org/wso2/carbon/event/admin/internal/TopicManagerAdminService.java

```diff
@@ -10,49 +12,102 @@
 
 import java.util.Calendar;
 
+/**
+ * Provides topic related functions as a web service.
+ */
 public class TopicManagerAdminService {
 
+    /**
+     * Logger to log information, warning, errors.
+     */
+    private static Log log = LogFactory.getLog(TopicManagerAdminService.class);
+
+    /**
+     * Gets all the topics
+     *
+     * @return A topic node
+     * @throws EventAdminException
+     */
     public TopicNode getAllTopics() throws EventAdminException {
         EventBroker eventBroker = EventAdminHolder.getInstance().getEventBroker();
         try {
             return eventBroker.getTopicManager().getTopicTree();
         } catch (EventBrokerException e) {
-            throw new EventAdminException("Error in accessing topic manager ", e);
+            log.error("Error in accessing topic manager", e);
+            throw new EventAdminException("Error in accessing topic manager", e);
         }
     }
 
+    /**
+     * Gets the permission roles for a topic
+     * Suppressing warning as this is used as a web service
+     *
+     * @param topic Topic name
+     * @return An array of TopicRolePermission
+     * @throws EventAdminException
+     */
+    @SuppressWarnings("UnusedDeclaration")
     public TopicRolePermission[] getTopicRolePermissions(String topic) throws EventAdminException {
         EventBroker eventBroker = EventAdminHolder.getInstance().getEventBroker();
         try {
             return eventBroker.getTopicManager().getTopicRolePermission(topic);
         } catch (EventBrokerException e) {
+            log.error("Error in accessing topic manager", e);
             throw new EventAdminException("Error in accessing topic manager", e);
```

Comment:
May be the error message can be moved to a string variable and reuse it without defining it twice.
Thanks